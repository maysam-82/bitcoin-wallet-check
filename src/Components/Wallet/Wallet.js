import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
	render() {
		const { balance } = this.props;
		return (
			<div>
				<h2 className="balance">Money Balance: {balance}</h2>
			</div>
		);
	}
}

const mapStateToProps = state => {
	return { balance: state };
};

export default connect(mapStateToProps)(Wallet);
