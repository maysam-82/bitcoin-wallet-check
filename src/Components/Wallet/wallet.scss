.wallet-container {
	max-width: 18.75rem;
	width: 100%;
	margin: 0.625rem auto;
	padding: 0.625rem;
	border: 0.0625rem solid #ccc;
	display: flex;
	flex-wrap: wrap;
	box-shadow: 0.25rem 0.25rem rgba($color: #000000, $alpha: 0.6);
	.balance {
		width: 100%;
		text-align: center;
		font-size: 1rem;
	}
	.input-container {
		width: 100%;
		margin: 0.625rem auto;
		.input-money {
			width: 100%;
			font-weight: bold;
		}
	}
	.button-container {
		width: calc(50% - 0.625rem);
		margin-right: 1.25rem;
		button {
			background-color: blue !important;
		}
		&:last-child {
			margin-right: 0;
			button {
				background-color: red !important;
			}
		}
		button {
			width: 100%;
			color: #fff;
			font-weight: bold;
		}
	}
}
