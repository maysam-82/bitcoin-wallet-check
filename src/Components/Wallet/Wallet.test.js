import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('Wallet Component', () => {
	const props = { balance: 20 };
	const wrapper = shallow(<Wallet {...props} />);
	it('should render Wallet component correctly', () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});
	it('should display the balance equal to 20', () => {
		expect(wrapper.find('.balance').text()).toEqual('Money Balance: 20');
	});
});
