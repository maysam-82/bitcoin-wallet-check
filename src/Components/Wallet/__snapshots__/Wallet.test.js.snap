// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Component should render Wallet component correctly 1`] = `
"<div>
  <h2 className=\\"balance\\">
    Money Balance: 
    20
  </h2>
</div>"
`;
